### PR TITLE
Remove semi-convoluted wizard & model structure from before-you-claim

### PR DESCRIPTION
**Pull request by wpears**
This results in no functional changes, just code reorganization/deletion.

Tests and functional tests should pass.

### CODE REVIEW

**Comment by anselmbradford** [comment]
Any interest in building out the tests in https://github.com/cfpb/consumerfinance.gov/blob/main/test/cypress/integration/consumer-tools/before-you-claim/before-you-claim.cy.js to check the slider, the Step 2 tips + popups, and Step 3 selection?

**Comment by anselmbradford** [comment]
> I think that would be good if we were planning to totally rewrite this tool and is probably even necessary to pull jquery out, but out of scope for this PR, imo.

Agreed it's out of scope for this PR, but in general it would be a nice to have to catch regressions in the tool, since I don't believe how it should work is monitored carefully.

**Comment by wpears** [comment]
hah! I tried to get too cute in the commit before this and removed some `substr` calls which SEEMED pointless..... 😅 Fixed and added a test for that behavior which now passes.

**Review by BrewCityBoy — APPROVED** [review]

File: test/unit_tests/apps/retirement/js/data/index-spec.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+fetchMock.enableMocks();
+
+describe('fetchApiData', function () {
+  it('fetches data', function () {
+    fetch.mockResponseOnce(JSON.stringify({ data: '12345' }));
+    fetchApiData('11-11-1965', 34567).then((res) => {
+      expect(res.data).toEqual('12345');
+    });
+  });
```
Comment (anselmbradford) [inline test/unit_tests/apps/retirement/js/data/index-spec.js:18]: Sonarcloud is flagging this. Looks like you need:

```js
it('fetches data', async function () {
    fetch.mockResponseOnce(JSON.stringify({ data: '12345' }));
    await fetchApiData('11-11-1965', 34567).then((res) => {
      expect(res.data).toEqual('12345');
    });
  });
```

File: cfgov/unprocessed/apps/retirement/js/views/graph-view.js [hunk 2]
```diff
@@ -338,20 +335,18 @@ function placeBenefitsText() {
...
-  const selectedAboveFRA = selectedAge > SSData.fullAge;
-  const selectedCurrent = selectedAge === SSData.currentAge;
-  // const isFRA = SSData.currentAge === SSData.fullAge;
-  // const isYoungerThanFRA = SSData.currentAge < SSData.fullAge;
+  const selectedBelowFRA = selectedAge < benefits.fullAge;
+  const selectedFRA = selectedAge === benefits.fullAge;
+  const selectedAboveFRA = selectedAge > benefits.fullAge;
+  const selectedCurrent = selectedAge === benefits.currentAge;
+  // const isFRA = benefits.currentAge === benefits.fullAge;
+  // const isYoungerThanFRA = benefits.currentAge < benefits.fullAge;
```
Comment (anselmbradford) [inline cfgov/unprocessed/apps/retirement/js/views/graph-view.js:349]: Can these be deleted?

Comment (wpears) [reply cfgov/unprocessed/apps/retirement/js/views/graph-view.js]: Deleted in 22f698dc9e1629f4ef613265b3abed74cb2a39eb

File: test/unit_tests/apps/retirement/js/data/index-spec.js (hunk 1 above)
Comment (wpears) [reply test/unit_tests/apps/retirement/js/data/index-spec.js]: Fixed in 22f698dc9e1629f4ef613265b3abed74cb2a39eb by explicitly handling the reject case

**Pull request merged**